### ISSUE TITLE
EF6 PK / Index annotation - duplicate index issue

### ISSUE THREAD

**Issue by seanlmcgill** [ReversePOCO/EntityFramework-Reverse-POCO-Code-First-Generator#587]
Was trying to do a quick eval/POC using this template along w/ EF6 Migrations and I ran into an issue where I'm getting a seemingly duplicate Index annotation on a PK field.  I'm using Visual Studio 2019 Professional, v3.1.3 of the template, .net 4.7.2, EF 6.4  Not sure if this is something I'm doing wrong, or it's actually coming out correct and there's another issue, etc:

I have a table defined as such (renamed it Thing for this example, as well as left most of the cols out):

```sql
CREATE TABLE [dbo].[Thing](
	[ThingId] [int] IDENTITY(1,1) NOT NULL,
	[Name] [nvarchar](100) NOT NULL,
	-- other cols left out...
 CONSTRAINT [PK_Thing] PRIMARY KEY CLUSTERED 
(
	[ThingId] ASC
)WITH (PAD_INDEX = OFF, STATISTICS_NORECOMPUTE = OFF, IGNORE_DUP_KEY = OFF, ALLOW_ROW_LOCKS = ON, ALLOW_PAGE_LOCKS = ON) ON [PRIMARY]
) ON [PRIMARY]
GO
```

Here's my basic template settings:
```c#
 Settings.DatabaseType            = DatabaseType.SqlServer; 
 Settings.TemplateType            = TemplateType.Ef6; 
 Settings.GeneratorType           = GeneratorType.Ef6;
 Settings.UseMappingTables        = true;
 Settings.FileManagerType         = FileManagerType.VisualStudio; 
 Settings.UseDataAnnotations      = true;
```

The generated code ends up adding the Index annotation to the PK field in the 'OnModelCreating' method: 

```c#
// Indexes        
            modelBuilder.Entity<Thing>()
                .Property(e => e.ThingId)
                .HasColumnAnnotation(
                    IndexAnnotation.AnnotationName,
                    new IndexAnnotation(new IndexAttribute("PK_Thing", 1) { IsUnique = true, IsClustered = true })
                );
```
It is also marked as [Key] on the POCO class:

```c#
public class Thing
    {
        [Key]
        [Required]
        [Display(Name = "Thing ID")]
        public int ThingId { get; set; } // ThingId (Primary key)
```

Up to that point all runs ok and such however, is this correct to have both the Index & Key annotation?  As next I do 'Add-Migration' and get the following migration code:

```c#
CreateTable(
                "dbo.Thing",
                c => new
                    {
                            ThingId = c.Int(nullable: false, identity: true),
			    // other fields omitted
                    })
                .PrimaryKey(t => t.ThingId)
                .Index(t => t.ThingId, unique: true, clustered: true, name: "PK_Thing");
```

The migration fails as it's trying to add both the clustered primary key field and also a separate clustered index of a very similar name which results in the error:

**System.Data.SqlClient.SqlException (0x80131904): Cannot create more than one clustered index on table 'dbo.Thing'. Drop the existing clustered index 'PK_dbo.Thing' before creating another.**
:frowning:

If i remove the 'Index(...' entry on the migration, it works fine - so I'm assuming it's not valid to have that field marked as both, ie - PK takes care of making it clustered and such.  So as a quick test, I modified the template to not add that index annotation if the index name starts w/ PK_ - and my issues went away.  (not sure if that's a great solution but rather a quick fix to see if my assumptions were correct).

Anyone else having this problem or am I just missing / doing something incorrectly? 
:thinking:

Any help is much appreciated!  

:smile:

**Comment by sjh37** [comment]
Hi @seanlmcgill 
You are correct. Thanks for spotting this!
All fixed, and will be in the next release.
For now, grab the latest code from [GitHub EF.Reverse.POCO.v3.ttinclude](https://raw.githubusercontent.com/sjh37/EntityFramework-Reverse-POCO-Code-First-Generator/ef6d10789216af4e3280ce64069a207c81b9c08c/EntityFramework.Reverse.POCO.Generator/EF.Reverse.POCO.v3.ttinclude)

**Comment by sjh37** [comment]
Any problems, please let me know and I shall re-open. The fix was to remove primary keys from the index generation section.
Thanks again Sean.

**Comment by seanlmcgill** [comment]
Awesome, thank you very much for the quick response!

**Comment by sjh37** [comment]
Now released as part of v3.2.0
